web/styles/message_row.css

```diff
@@ -105,9 +105,7 @@ $time_column_min_width: 42px; /* + padding */
         padding: 0 5px 0 3px;
 
         &:hover .message_controls,
-        &:focus-within .message_controls,
-        &:hover .message_failed,
-        &:focus-within .message_failed {
+        &:hover .message_failed {
```

Comment:
4cd23103db2172aed1594a3e958a1a8eefe3631f looks like what added this code; so yeah it was motivated by accessibility concerns but not complete.